src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/PackageDeclarationCheck.java

```diff
@@ -30,21 +33,39 @@
  *
```

Comment:
Description few lines above should be changed to include checking directory structure like xdoc does.
Would be also good to add the rationale section.